core/src/test/kotlin/fr/sncf/osrd/stdcm/preprocessing/BlockAvailabilityTests.kt

```diff
@@ -636,4 +636,330 @@ class BlockAvailabilityTests {
+     * train pass after another step: should return Unavailable(infinity, ...).
+     */
+    @Test
+    fun testPassingBeforeStepCreatesDelayMakesAvailableStepUnavailable() {
+        val explorer = makeExplorer(5, 1)
+        val plannedStepOffset = Offset<Block>(0.meters)
+        val secondPlannedStepOffset = Offset<Block>(100.meters)
+        val timeAtZoneEnd = explorer.interpolateDepartureFromClamp(explorer.getSimulatedLength())
+        val steps =
+            listOf(
+                // We need to add delay = timeAtZoneEnd
+                STDCMStep(
+                    listOf(PathfindingEdgeLocationId(blocks[0], plannedStepOffset)),
+                    null,
+                    false,
+                    PlannedTimingData(timeAtZoneEnd.seconds, 0.seconds, 0.seconds)
+                ),
+                // If we add delay = timeAtZoneEnd, this step is not respected
+                STDCMStep(
+                    listOf(PathfindingEdgeLocationId(blocks[0], secondPlannedStepOffset)),
+                    null,
+                    false,
+                    PlannedTimingData(0.seconds, 0.seconds, timeAtZoneEnd.seconds)
+                )
+            )
+        val availability = makeBlockAvailability(infra, listOf(), listOf(), steps)
+        val res =
+            availability.getAvailability(
+                explorer,
+                Offset(0.meters),
+                explorer.getSimulatedLength(),
+                0.0
+            ) as BlockAvailabilityInterface.Unavailable
+        // The delay added to solve the first step makes the train pass after the second step =>
+        // infinity + second step offset
+        assertEquals(POSITIVE_INFINITY, res.duration)
+        assertEquals(plannedStepOffset.distance, res.firstConflictOffset.distance)
+    }
+
+    /** Unavailable step has priority over available conflicts. */
+    @Test
+    fun testPassingBeforePlannedStepWithoutUnavailableConflicts() {
+        val explorer = makeExplorer(5, 1)
+
+        val plannedStepOffset = Offset<Block>(100.meters)
+        val timeAtStep = explorer.interpolateDepartureFromClamp(Offset(plannedStepOffset.distance))
+        val timeAtZoneEnd = explorer.interpolateDepartureFromClamp(explorer.getSimulatedLength())
+        val steps =
+            listOf(
+                STDCMStep(
+                    listOf(PathfindingEdgeLocationId(blocks[0], Offset(100.meters))),
+                    null,
+                    false,
+                    PlannedTimingData(timeAtZoneEnd.seconds, 0.seconds, 30.seconds)
+                )
+            )
+        val stepUnavailableDuration = timeAtZoneEnd.seconds.seconds - timeAtStep
+
+        val requirements =
+            listOf(
+                // Requirement ending just before step becomes available
+                SpacingRequirement(zoneNames[0], 0.0, stepUnavailableDuration, true),
+                // Requirement starting just after step becomes available
```

Comment:
```suggestion
                // Requirement starting just after step becomes unavailable
```

Either a typo or the code doesn't match

Reply:
Hmm. Not exactly. The unavailable step becomes available if you delay it by `stepUnavailableDuration`. As a result, this conflict, starting at `timeAtZoneEnd + stepUnavailableDuration` does indeed start just after the step becomes available if we add delay. I could put `step becomes available after adding stepUnavailableDuration as delay`.

Reply:
It still feels wrong, the step becomes available at `timeAtZoneEnd`, not `timeAtZoneEnd - timeAtStep`. This isn't related to the traversal time, it should be an absolute time (if the names match)

Reply:
Seen with @eckter. This test is quite hard to undertand, and the test case is already tested by `testPassingBeforePlannedStep`: test was removed.